smserver/smutils/smpacket/smpacket.py

```diff
@@ -511,6 +511,10 @@ class SMPacketClientNSCGSR(SMPacket):
         (smencoder.SMPayloadType.NT, "song_options", None),
         (smencoder.SMPayloadType.NT, "first_player_options", None),
         (smencoder.SMPayloadType.NT, "second_player_options", None),
+        (SMPayloadType.NT, "first_player_chartkey", None),
+        (SMPayloadType.NT, "second_player_chartkey", None),
+        (SMPayloadType.INT, "rate", 0),
+        (SMPayloadType.NT, "filehash", None)
```

Comment:
Just did so. I'm not sure if i implemented the changes to the test properly(I jut added some example values for the new fields in https://github.com/ningirsu/stepmania-server/blob/master/test/test_smutils/test_smpacket.py#L65)